manager/controllers/packages.go

```diff
@@ -30,16 +29,34 @@ type PackagesResponse struct {
 	Meta  ListMeta      `json:"meta"`
 }
 
-func packagesQuery(acc int) *gorm.DB {
-	return database.Db.
-		Select(PackagesSelect).
+// Used as a for subquery performing the actual calculation which is joined with latest summaries
+type queryItem struct {
+	NameID           int    `query:"p.name_id"`
+	Name             string `json:"name" query:"pn.name"`
+	SystemsInstalled int    `json:"systems_installed" query:"count(sp.id)"`
+	SystemsUpdatable int    `json:"systems_updatable" query:"count(sp.id) filter (where spkg.update_data is not null)"`
+}
+
+var queryItemSelect = database.MustGetSelect(&queryItem{})
+
+func packagesQuery(c *gin.Context, acc int) *gorm.DB {
+	subQ := database.Db.
+		Select(queryItemSelect).
 		Table("system_platform sp").
-		Joins("inner join rh_account ra on sp.rh_account_id = ra.id").
-		Joins("inner join system_package spkg on spkg.system_id = sp.id").
+		// nolint: lll
+		Joins("inner join system_package spkg on spkg.system_id = sp.id and sp.stale = false and sp.rh_account_id = ?", acc).
 		Joins("inner join package p on p.id = spkg.package_id").
 		Joins("inner join package_name pn on pn.id = p.name_id").
 		Where("spkg.rh_account_id = ?", acc).
-		Group("ra.id, pn.id, pn.name")
+		Group("p.name_id, pn.name")
```

Comment:
Have you tested this request in CI or prod db? Woudn't it helped to add "where sp.rh_account_id = acc".

Reply:
Added.